modules/line_descriptor/src/binary_descriptor.cpp

```diff
@@ -1347,6 +1349,9 @@ int BinaryDescriptor::computeLBD( ScaleLines &keyLines, bool useDetectionData )
     for ( int g = 0; g < 32; g++ )
     {
       /* get LBD data */
+      if (keyLines[lineIDInScaleVec].empty()) {
```

Comment:
It looks like this loop is used to fill `appogio` matrix. However it is a local matrix that will be discarded just after the loop. Is it necessary to fill it even if  `keyLines[lineIDInScaleVec]` isn't empty?

Reply:
Perhaps initialization with zero values should work:
```
cv::Mat appoggio = cv::Mat(1, 32, CV_32FC1, Scalar::all(0));
```

Reply:
I reviewed the code again, and believe @terfendail is right. I removed the whole code block including appogio matrix and tested, it works perfectly.